File: Ruby/design_patterns/src/facade/observable.rb
class Observable

  def initialize
    @length = 0
    @modifiers = Array.new
    @observers = Array.new
  end

  def increment_length
    @length = @length +1
  end

  def add_observer(observer)
    @observers.push(observer)
  end

  def remove_observer(observer)
    @observers.delete(observer)
  end

  def trigger_notification
    notify_all
  end

  def add_modifier(name)
    @modifiers.push(name)
  end

  def notify_all
    @observers.each do |observer|
      observer.receive_notification('A notification was sent')
    end
  end

  def get_observers
    @observers
  end

  def get_length
    @length
  end
end

File: JavaScript/design_patterns/src/facade/observable.js
var Observer = require('./observer').Observer;

exports.Observable = function () {
    'use strict';

    var size = 0;
    var observers = [];
    var modifiers = {};

    this.addObserver = function addObserver(observer) {
        if (observer instanceof Observer) {
            observers.push(observer);
        }
        return observers;
    };

    this.removeObserver = function removeObserver(observer) {
        if (observer instanceof Observer) {
            var index = observers.indexOf(observer);
            if (index > -1) {
                observers.splice(index);
            }
        }
    };

    this.increaseSize = function () {
        size++;
    };

    this.notifyAll = function notifyAll() {
        var index = 0,
            value = '';

        for (index; index < observers.length; index++) {
            value = observers[index];
            if (value instanceof  Observer) {
                value.addNotification('A new notification was received');
            }
        }
    };

    this.triggerChange = function () {
        notifyAll();
    };

    this.getSize = function gerSize() {
        return size;
    };

    this.getObservers = function () {
        return observers;
    };

    this.getModifiers = function () {
        return modifiers;
    };
};